### PR TITLE
Enable the pg_regress tests on staging for PG17

### PR DESCRIPTION
**Pull request by a-masterov**
## Problem
Currently, we run the `pg_regress` tests only for PG16
However, PG17 is a part of Neon and should be tested as well 
## Summary of changes
Modified the workflow and added a patch for PG17 enabling the `pg_regress` tests.
The problem with leftovers was solved by using branches.

### CODE REVIEW

File: .github/workflows/cloud-regress.yml [hunk 1]
```diff
@@ -73,15 +83,68 @@ jobs:
...
+          project_id: ${{matrix.project-id}}
+          endpoint_id: ${{steps.endpoint-id.outputs.id}}
+          branch_id: ${{steps.create-branch.outputs.branch_id}}
+
+      - name: Wait for operations to be completed
+        uses: ./.github/actions/neon-wait-operations-to-finish
+        with:
+          api_key: ${{ secrets.NEON_STAGING_API_KEY }}
+          project_id: ${{matrix.project-id}}
+
```
Comment (bayandin) [inline .github/workflows/cloud-regress.yml:122]: I do not fully understand why `create-branch` only is not enough here, and why do we need to re-attach endpoints?

Comment (a-masterov) [reply .github/workflows/cloud-regress.yml:122]: It's a good question, thanks, let me try to explain.
We have to add some settings to the endpoint (e.g. `TimeZone`, `compute_query_id` to `off`). We cannot add them by the API call due to restrictions (or not yet implemented). So updating the control plane database is the only way. But we cannot do this from our workflows, so we have to attach the pre-created endpoint to the new branch before testing and re-attach it to the main branch after.

Comment (a-masterov) [reply .github/workflows/cloud-regress.yml]: Discovered `default_endpoint_settings`, moved to using it

File: .github/workflows/cloud-regress.yml [hunk 2]
```diff
@@ -73,15 +79,42 @@ jobs:
...
+        uses: ./.github/actions/neon-branch-create
+        with:
+          api_key: ${{ secrets.NEON_STAGING_API_KEY }}
+          project_id: ${{ vars[format('PGREGRESS_PG{0}_PROJECT_ID', matrix.pg-version)] }}
+
+      - name: Wait for operations to be completed
+        uses: ./.github/actions/neon-wait-operations-to-finish
+        with:
+          api_key: ${{ secrets.NEON_STAGING_API_KEY }}
+          project_id: ${{ vars[format('PGREGRESS_PG{0}_PROJECT_ID', matrix.pg-version)] }}
```
Comment (bayandin) [inline .github/workflows/cloud-regress.yml:93]: That's already done in `neon-branch-create` (or it should be done there)

File: .github/workflows/cloud-regress.yml [hunk 3]
```diff
@@ -73,15 +79,42 @@ jobs:
...
+
+      - name: Get the connection URI
+        id: connect-uri
+        uses: ./.github/actions/neon-get-connection-uri
+        with:
+          api_key: ${{ secrets.NEON_STAGING_API_KEY }}
+          project_id: ${{ vars[format('PGREGRESS_PG{0}_PROJECT_ID', matrix.pg-version)] }}
```
Comment (bayandin) [inline .github/workflows/cloud-regress.yml:100]: `neon-branch-create` returns new connections string, can we use it?

File: .github/workflows/cloud-regress.yml [hunk 4]
```diff
@@ -73,15 +79,42 @@ jobs:
...
           build_type: ${{ env.BUILD_TYPE }}
           test_selection: cloud_regress
-          pg_version: ${{ env.DEFAULT_PG_VERSION }}
+          pg_version: ${{matrix.pg-version}}
           extra_params: -m remote_cluster
         env:
-          BENCHMARK_CONNSTR: ${{ secrets.PG_REGRESS_CONNSTR }}
+          BENCHMARK_CONNSTR: ${{steps.connect-uri.outputs.uri}}
+
+      - name: Delete branch
```
Comment (bayandin) [inline .github/workflows/cloud-regress.yml:112]: `if: always()`?

File: .github/workflows/cloud-regress.yml [hunk 5]
```diff
@@ -73,15 +79,42 @@ jobs:
           path: /tmp/neon/
           prefix: latest
 
+      - name: Create a new branch
+        id: create-branch
+        uses: ./.github/actions/neon-branch-create
+        with:
+          api_key: ${{ secrets.NEON_STAGING_API_KEY }}
+          project_id: ${{ vars[format('PGREGRESS_PG{0}_PROJECT_ID', matrix.pg-version)] }}
```
Comment (bayandin) [inline .github/workflows/cloud-regress.yml:87]: I liked one of the previous approaches with `${{ vars.PGREGRESS_PG16_PROJECT_ID }}` listed in matrix. I think the readability was better then

File: .github/workflows/cloud-regress.yml (hunk 2 above)
Comment (a-masterov) [reply .github/workflows/cloud-regress.yml]: Yes, you're right, done.

File: .github/workflows/cloud-regress.yml (hunk 5 above)
Comment (a-masterov) [reply .github/workflows/cloud-regress.yml:87]: I agree about readability but in this case, the redundant project name is shown with the version.

File: .github/workflows/cloud-regress.yml (hunk 3 above)
Comment (a-masterov) [reply .github/workflows/cloud-regress.yml]: Yes, we can

File: .github/actions/neon-wait-operations-to-finish/action.yml
Comment (bayandin) [inline .github/actions/neon-wait-operations-to-finish/action.yml:1]: Let's delete this one as well

**Review by bayandin — APPROVED** [review]
Looks good!
Need to delete one more extra file, and we're good to merge it! Well done!

File: .github/actions/neon-wait-operations-to-finish/action.yml
Comment (a-masterov) [reply .github/actions/neon-wait-operations-to-finish/action.yml:1]: Done, thanks

**Pull request merged**